core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1704,7 +1704,7 @@ public FetchDataInfo read(RemoteStorageFetchInfo remoteStorageFetchInfo) throws
                     new LogOffsetMetadata(offset, remoteLogSegmentMetadata.startOffset(), startPos),
                     MemoryRecords.readableRecords(buffer));
             if (includeAbortedTxns) {
-                fetchDataInfo = addAbortedTransactions(firstBatch.baseOffset(), remoteLogSegmentMetadata, fetchDataInfo, logOptional.get());
+                fetchDataInfo = addAbortedTransactions(firstBatch.baseOffset(), remoteLogSegmentMetadata, fetchDataInfo, logOptional.get(), enrichedRecordBatch.getSkippedBytes());
```

Comment:
Instead of passing in skippedBytes, could we just add it to startPos? Also, when creating FetchDataInfo, it's more accurate to use firstBatch.baseOffset instead of offset.

Reply:
done.